src/Evolution/Systems/NewtonianEuler/TimeDerivative.hpp

```diff
@@ -0,0 +1,150 @@
+      const tnsr::I<DataVector, Dim>& velocity,
+      const Scalar<DataVector>& pressure, const SourceTerm& source,
+      const SourceTermArgs&... source_term_args) noexcept {
+    detail::fluxes_impl(mass_density_cons_flux, momentum_density_flux,
+                        energy_density_flux, enthalpy_density, momentum_density,
+                        energy_density, velocity, pressure);
+    if constexpr (not std::is_same_v<SourceTerm, Sources::NoSource>) {
+      sources_impl(
+          std::make_tuple(dt_mass_density, dt_momentum_density,
+                          dt_energy_density),
+          typename InitialDataType::source_term_type::sourced_variables{},
+          source, source_term_args...);
+    }
+  }
+
+  template <typename... SourceTermArgs>
+  static void apply(
+      // Time derivatives returned by reference. No source terms or
+      // nonconservative products, so not used. All the tags in the
+      // variables_tag in the system struct.
+      const gsl::not_null<Scalar<DataVector>*> /*dt_mass_density*/,
+      const gsl::not_null<tnsr::I<DataVector, Dim>*> /*dt_momentum_density*/,
+      const gsl::not_null<Scalar<DataVector>*> /*dt_energy_density*/,
+
+      // Fluxes returned by reference. Listed in the system struct as
+      // flux_variables.
+      const gsl::not_null<tnsr::I<DataVector, Dim>*> mass_density_cons_flux,
+      const gsl::not_null<tnsr::IJ<DataVector, Dim>*> momentum_density_flux,
+      const gsl::not_null<tnsr::I<DataVector, Dim>*> energy_density_flux,
+
+      // Temporaries returned by reference. Listed in temporary_tags above.
+      const gsl::not_null<Scalar<DataVector>*> enthalpy_density,
+
+      // Arguments listed in argument_tags above
+      const tnsr::I<DataVector, Dim>& momentum_density,
+      const Scalar<DataVector>& energy_density,
+      const tnsr::I<DataVector, Dim>& velocity,
+      const Scalar<DataVector>& pressure) noexcept {
+    detail::fluxes_impl(mass_density_cons_flux, momentum_density_flux,
+                        energy_density_flux, enthalpy_density, momentum_density,
+                        energy_density, velocity, pressure);
+  }
+
+ private:
+  using dt_vars_list =
+      tmpl::list<Tags::MassDensityCons, Tags::MomentumDensity<Dim>,
+                 Tags::EnergyDensity>;
+
+  template <typename... SourceTermArgs, typename... SourcedVars>
+  static void sources_impl(std::tuple<gsl::not_null<Scalar<DataVector>*>,
+                                      gsl::not_null<tnsr::I<DataVector, Dim>*>,
+                                      gsl::not_null<Scalar<DataVector>*>>
+                               dt_vars,
+                           tmpl::list<SourcedVars...> /*meta*/,
+                           const SourceTerm& source,
+                           const SourceTermArgs&... source_term_args) noexcept {
+    source.apply(
+        std::get<tmpl::index_of<dt_vars_list, SourcedVars>::value>(dt_vars)...,
+        source_term_args...);
+  }
```

Comment:
I agree. I think this is fallout from what I'll call "over DataBoxification" (aka too many compute tags). I think if instead we compute everything locally in the fluxes and sources then this could be reworked and simplified. Luckily, this is the direction we're heading anyway :)